src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.Linux.cs

```diff
@@ -290,25 +332,49 @@ protected internal override WaitForEventResult WaitForEvent(int pinNumber, PinEv
             AddPinToPoll(pinNumber, ref valueFileDescriptor, ref pollFileDescriptor, out bool closePinValueFileDescriptor);
 
             bool eventDetected = WasEventDetected(pollFileDescriptor, valueFileDescriptor, out _, cancellationToken);
+            if (_statusUpdateSleepTime > 0)
+            {
+                Thread.Sleep(_statusUpdateSleepTime); // Adding some delay to make sure that the value of the File has been updated so that we will get the right event type.
```

Comment:
Is this trying to protect to multi-threaded aps? because if so I think that this won't really work 100% of the time.

Reply:
oh, actually I see what the idea is here. I guess if our event is detected too fast you may have the case that the value file hasn't happened, is that possible with the Sysfs driver though? I thought that SysFS you are basically locking on that file change, so you would only really detect an event if the value of the file actually changes, plus I would imagine that the Read() operation alone would take a few ms which would give some extra time for the value to be right. Since you have done more testing, feel free to correct me if I'm wrong here.

Reply:
This may now look weird, yes. In fact, I copied this from line 574 (method DetectEvents()) where a comment says that this is needed. I didn't want to change the SysFs implementation when used standalone, so I first just made sure the two equivalent methods are implemented similarly. 

As a second step then, I reduced the _statusUpdateSleepTime to 0 for the InterruptSysFsDriver, because there the Read() operation uses the registers and is therefore faster and never needs a wait. 

If you want, I can do another test whether that delay is required for SysFs alone. Maybe that comment there on line 574 is just wrong.